src/Framework/PaymentGateways/PaymentGatewayRegister.php

```diff
@@ -25,6 +26,21 @@ public function getPaymentGateways(): array
         return $this->gateways;
     }
 
+    /**
+     * Get NextGen Gateways
+     *
+     * @unreleased
+     */
+    public function getNextGenPaymentGateways(): array
+    {
+        return array_filter(
+            $this->gateways,
+            static function ($gateway) {
+                return is_a($gateway, NextGenPaymentGatewayInterface::class, true);
+            }
+        );
+    }
+
```

Comment:
@pauloiankoski the NextGen terminology is specific to our feature plugin so we don't want this to live in our core gateway registrar. Currently the `NextGenPaymentGatewayInterface` lives in the feature plugin as well.  Eventually we'll rename that and it will all live in core.

For now, you can just add this logic as a local method or a repository method within the feature plugin

Reply:
Reverted 228518983269eb56f26a89f0a90bec128232500d